prow/cmd/autobump/bump.sh

```diff
@@ -66,34 +66,52 @@ main() {
   else
     usage
   fi
-  echo -e "Bumping: 'gcr.io/k8s-prow/' images to $(color-version ${new_version}) ..." >&2
+  echo -e "Bumping: 'gcr.io/k8s-prow/' images to $(color-version "${new_version}") ..." >&2
 
   bumpfiles=($(add_suffix "$(split_on_commas "$COMPONENT_FILE_DIR")"))
   bumpfiles+=("${CONFIG_PATH}")
   if [[ -n "${JOB_CONFIG_PATH}" ]]; then
     bumpfiles+=($(grep -rl -e "gcr.io/k8s-prow/" "${JOB_CONFIG_PATH}"))
   fi
 
-  echo -e "Attempting to bump the following files: ${bumpfiles[@]}" >&2
+  echo -e "Attempting to bump the following files:" >&2
+  IFS=$'\n'; echo -e "${bumpfiles[*]}" >&2
 
+  local token="$(gcloud config config-helper --format 'value(credential.access_token)')"
   # Update image tags in the identified files.
-  filter="s/gcr.io\/k8s-prow\/\([[:alnum:]_-]\+\):v[a-f0-9-]\+/gcr.io\/k8s-prow\/\1:${new_version}/I"
+  local matcher="gcr.io\/k8s-prow\/\([[:alnum:]_-]\+\):v[a-f0-9-]\+"
+  local replacer="s/${matcher}/gcr.io\/k8s-prow\/\1:${new_version}/I"
   for file in "${bumpfiles[@]}"; do
-    ${SED} -i "${filter}" ${file}
+    local images="$(grep -o "${matcher}" "${file}")"
+    local array=(${images//\\n/})
+    for i in "${!array[@]}"; do
```

Comment:
No need for the index you're not doing anything with: `for image in "${array[@]}"; do`

Reply:
I will get `array[*]: unbound variable` after changing to `for image in "${array[@]}"` when there is no matched image in the file. Not sure how to deal with it so I just kept as it is.

Reply:
This works just fine for me:


```sh
 set -o errexit
 set -o nounset
 set -o pipefail
 shopt -s dotglob
 shopt -s nullglob
 
 array=()
 for image in "${array[@]}"; do
     echo hi $image
 done
```

Reply:
It seems for different bash versions, the behavior is different for handling empty arrays.
I have changed to `for image in ${arr[@]+"${arr[@]}"};` as suggested in https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u, and it works.